### PR TITLE
chore: [util/dex|glob] Add security logging

### PR DESCRIPTION
**Pull request by my-git9**
[util/dex|glob] Add security logging

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by jannfis** [comment]
Thanks @my-git9 

Can you please elaborate a little on the security impact of these errors, i.e. why should they emit a security log event? Thanks!

**Comment by my-git9** [comment]
> Thanks @my-git9
> 
> Can you please elaborate a little on the security impact of these errors, i.e. why should they emit a security log event? Thanks!
Discussions and PRs about security logs are here, please see
https://github.com/argoproj/argo-cd/pull/10361#discussion_r949187640
https://github.com/argoproj/argo-cd/pull/10256

**Comment by jannfis** [comment]
Yes, thanks, I'm aware of this. I was just wondering how the parts of the code you are adding it to are relevant to emit security logging? 

Just trying to understand the reasoning behind this particular change.

**Comment by notfromstatefarm** [comment]
The dex error I can see being useful for security logging - dex is a wholly security-related service. However, logging any time a glob fails to compile anywhere in our codebase doesn't seem security-related.

**Comment by my-git9** [comment]
I see, I'll tweak the code a bit

**Comment by my-git9** [comment]
@notfromstatefarm @jannfis I'm sorry, I don't know enough about security logs. Can you think of this?

File: util/dex/dex.go [hunk 1]
```diff
@@ -83,7 +84,10 @@ func NewDexHTTPReverseProxy(serverAddr string, baseHRef string, tlsConfig *DexTL
 			if err != nil {
 				return err
 			}
-			log.Errorf("received error from dex: %s", string(b))
+			log.WithFields(log.Fields{
+				common.SecurityField:    common.SecurityMedium,
+				common.SecurityCWEField: 775,
```
Comment (crenshaw-dev) [inline util/dex/dex.go:89]: @my-git9 can you drop this line? CWE-775 is specific to failed file closing. I'm not sure there's a specific CWE that makes sense for an error message from Dex.

**Review by crenshaw-dev — APPROVED** [review]
Thanks @my-git9!

**Pull request merged**